### ISSUE TITLE
Support CalDAV tasks

### ISSUE THREAD

**Issue by PanderMusubi** [FossifyOrg/Notes#36]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Notes/issues) or [closed](https://github.com/FossifyOrg/Notes/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Notes/discussions) or [closed](https://github.com/FossifyOrg/Notes/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Notes/blob/master/CONTRIBUTING.md).
- [X] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

Support syncing the notes as CalDAV tasks.

### Why do you want this feature?

Notes can be used in sync on mutiple devices.

### Additional information

See also https://f-droid.org/en/packages/org.tasks/

**Comment by Albirew** [comment]
I agree. ideally, it would allow for notes and tasks sync with home cloud solutions like nextcloud.
Tasks-org is a gas factory and opentasks is not maintained since 2021

**Comment by PanderMusubi** [comment]
With [DAVx⁵](https://www.davx5.com/) also possible to connect e.g. with [DAViCal](https://davical.org/) and more.

**Comment by daffydock** [comment]
Agreed. Using DAVx5 should be a far easier implementation that could cover a number of services, including Nextcloud and others.

**Comment by Kevinf63** [comment]
Ideally a clear distinction of VTodo and VJournal specs akin to JTX Board.

**Comment by naveensingh** [comment]
It looks like many people want this feature. "Problems" I see:

 - This is also requested in the [Calendar](https://github.com/FossifyOrg/Calendar/issues/36) app (and I think it's more appropriate there)
 - If this is done, sooner or later someone's going to request for task repetition (also available in Calendar)
 - Fossify Notes is supposed to be the simple(er) notes/tasks app.

I'm not dismissing this request but I'm also not sure if it aligns with the app's _current_ goal.

**Comment by Albirew** [comment]
i see your point...
i'm mainly using synced tasks for groceries and simple checklists, but as you say, it may be better to have tasks in a separate app instead of inside note app...

**Comment by PanderMusubi** [comment]
When using [Getting Things Done](https://en.wikipedia.org/wiki/Getting_Things_Done) it is considered bad practice to add a date time to a task. In that case, better create an event in your calendar. IMHO tasks in a separate task app should have no date time functionality, even though the ICS file format supports that. You would like to guard the scope of task and calendar app.

If people really need it, they could be helped with a function that converts a task in the task app into a calendar event in the calendar app and deletes the task afterwards, in the task app. Similarly, a calendar event could be converted into a task from the calendar app to the task app.

**Comment by komali2** [comment]
> When using [Getting Things Done](https://en.wikipedia.org/wiki/Getting_Things_Done) it is considered bad practice to add a date time to a task. In that case, better create an event in your calendar. IMHO tasks in a separate task app should have no date time functionality, even though the ICS file format supports that. You would like to guard the scope of task and calendar app.
> 
> If people really need it, they could be helped with a function that converts a task in the task app into a calendar event in the calendar app and deletes the task afterwards, in the task app. Similarly, a calendar event could be converted into a task from the calendar app to the task app.

Your second paragraph is essentially how Tasks.org works: Caldav VTODO items (tasks) can be selected and then you can choose "Create event from task" to get a date/time sensitive event (copying any start/end/date data to the event, if exists, or you can create it then). However, neither Tasks.org, nor JTXBoard, so far as I'm aware, allow for laying out VTODO items with date/time data on a calendar or agenda, as is possible in applications like Thunderbird (best) or Korganizer (some issues but basically works). 

Regarding your first paragraph, before the palm pilot and google calendar and GTD, people have been scheduling their tasks in hour-to-hour paper agendas. Here's mine for today:

![Image](https://github.com/user-attachments/assets/94823a47-2b1c-47c4-a815-83966a157851)

For 8 years I used this kind of task management system: calendar-focused, with backup todo lists. Then, I read the GTD book twice, and for 4 years switched to that + org mode, orgzly etc. I followed GTD to the letter, and I think it's a very helpful system, and parts of it I'll incorporate probably permanently into my task management system (e.g. the concept of need-to-file items TODO, filed items NEXT, projects, etc). However, it's not the last word on task management, and I've found that for ADHD, or at least my flavor of it, it's an ineffective system compared to time-based task management. 

A typical negative attribute of ADHD is difficulty realistically estimating tasks. A GTD system can thus quickly become overwhelming, opening my org agenda view for this context (work, home, whatever) sorted by priority containing 5 priority A tasks, 20 priority B, 45 priority C, etc, and this with weekly cleanups. And then a priority A task gets bumped week to week to week because new priority A tasks get created... etc. 

Compare that to a system where I sit down in the morning, take my task list, unschedule anything scheduled yesterday that didn't get done, and get to scheduling the day and comparing to the rest of the week, and now I can have a much more realistic view of not only priority, but capability (ostensibly what GTD was meant to solve iirc). Now I can say, actually that task needs to fit into this slot today. Now I can schedule pomodoros if I want, or walks, or other techniques important for ADHD people to actually get things done. 

TLDR task apps should have date time functionality because that would be useful to some people, and those that don't want this feature can simply not use it. Personally I'm very keen for an android application that allows me to see my tasks laid out on a calendar the way I can see them in Thunderbird and my paper agenda. Yes, a workaround is to just create events like "Work on xyz TODO," but integrating tasks directly on a calendar view is more useful, intuitive, and frictionless for me.  Especially since VTODO items can have subtasks that work decently in many clients, this would be nice. Many project management solutions visualize activities this way, in Gantt views or similar.

**Comment by PanderMusubi** [comment]
Thanks for.your insightful contribution.

Another way to go around it, for those who keeps tasks without datetime, is an option to disable/enable all the datetime functionality in the app and hide/show those controls in the GUI. Then we have the best of both worlds.

**Comment by dbrand666** [comment]
@komali2  YES!

You've laid out a lot of great use cases for the tasks integration.

The BEST mobile calendar app I've ever used was DateBk on PalmOS. It's the single thing I missed most when I moved to Windows Mobile and then to Android. I finally found what I needed in CalenGoo. An Android version of DateBk appeared eventually but I never put the effort into figuring out if I could switch back to it.

CalenGoo uses DavX5 calendars but seems to use its own sync for tasks. That's a little weird and needs extra setup and is presumably less efficient but it does get the job done. It has many options to customize just how the tasks are integrated into the various calendar views and can even show timed tasks at the right time integrated into your calendar. Tasks without a time show up too, either at the top or bottom of the day.

You can also turn off the display of tasks on calendar views.

I still love CalenGoo but am always looking for FOSS alternatives which is what brought me here today. I tried both E-tar and Fossify calendars and deleted both when I saw they lacked support for tasks.

OMG. I just noticed this issue was reported against the Notes app. There's a parallel issue reported against Calendar, which is probably where this discussion belongs. Confusingly, both are #36.

See https://github.com/FossifyOrg/Calendar/issues/36

**Comment by naveensingh** [comment]
Closing because of scope creep mentioned in https://github.com/FossifyOrg/Notes/issues/36#issuecomment-2628694250. This app will focus on note-taking and ephemeral todo items.
